container/docker/factory.go

```diff
@@ -139,6 +139,12 @@ func FullContainerName(dockerId string) string {
 func (self *dockerFactory) CanHandleAndAccept(name string) (bool, bool, error) {
 	// docker factory accepts all containers it can handle.
 	canAccept := true
+
+	// When using SystemD all docker containers have a docker- prefix
+	if UseSystemd() && !strings.HasPrefix(path.Base(name), "docker-") {
+		return false, canAccept, nil
```

Comment:
You are correct. I've not used that feature before :) So I just did a simple test and the prefix `docker-` is indeed replaced with the parent name. But it also didn't show up as a docker container in a build of master though. Probably because the docker factory is already relying on the `docker-` prefix when Systemd is detected :(